subsys/net/ip/net_if.c

```diff
@@ -4023,6 +4026,104 @@ static inline bool is_iface_offloaded(struct net_if *iface)
 		net_if_is_socket_offloaded(iface));
 }
 
+static void notify_iface_up(struct net_if *iface)
+{
+	net_if_flag_set(iface, NET_IF_RUNNING);
+	net_mgmt_event_notify(NET_EVENT_IF_UP, iface);
+
+
+	/* If the interface is only having point-to-point traffic then we do
+	 * not need to run DAD etc for it.
+	 */
+	if (!is_iface_offloaded(iface) &&
+	    !(l2_flags_get(iface) & NET_L2_POINT_TO_POINT)) {
+		iface_ipv6_start(iface);
+		net_ipv4_autoconf_start(iface);
+	}
+}
+
+static void notify_iface_down(struct net_if *iface)
+{
+	net_if_flag_clear(iface, NET_IF_RUNNING);
+	net_mgmt_event_notify(NET_EVENT_IF_DOWN, iface);
+
+	if (!is_iface_offloaded(iface) &&
+	    !(l2_flags_get(iface) & NET_L2_POINT_TO_POINT)) {
+		net_ipv4_autoconf_reset(iface);
+	}
+}
+
+static inline const char *net_if_oper_state2str(enum net_if_oper_state state)
+{
+	switch (state) {
+	case NET_IF_OPER_UNKNOWN:
+		return "UNKNOWN";
+	case NET_IF_OPER_NOTPRESENT:
+		return "NOTPRESENT";
+	case NET_IF_OPER_DOWN:
+		return "DOWN";
+	case NET_IF_OPER_LOWERLAYERDOWN:
+		return "LOWERLAYERDOWN";
+	case NET_IF_OPER_TESTING:
+		return "TESTING";
+	case NET_IF_OPER_DORMANT:
+		return "DORMANT";
+	case NET_IF_OPER_UP:
+		return "UP";
+	default:
+		break;
+	}
+
+	return "<invalid>";
```

Comment:
Perhaps return string values only if debugging is enabled?
If not then, we could just return empty value ""